Allow players in Client Admin to invite players to a Clan over the limit.

StudentAlleg opened the pull request:
Allows a Client Admin to invite a player to a Clan over the limit. Allows for players to join (if inivited) to a Clan that is over the limit but has a client admin in it.

Fixes #901

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] I have tested my changes.

StudentAlleg committed:
#901 - Allow players in Client Admin to invite players to a Clan over the limit.

Mykindos commented on clans/src/main/java/me/mykindos/betterpvp/clans/clans/listeners/ClanEventListener.java:
```diff
@@ -389,9 +389,27 @@ public void onJoinClanEvent(final MemberJoinClanEvent event) {
                 return;
             }
 
+            boolean hasAdministratingClient = false;
+            for (ClanMember clanMember : clan.getMembers()) {
+                final Optional<Client> clientOpt = clientManager.search().online(UUID.fromString(clanMember.getUuid()));
+                if (clientOpt.isPresent()) {
+                    final Client online = clientOpt.get();
+                    if (online.isAdministrating()) {
+                        hasAdministratingClient = true;
+                        break;
+                    }
+                }
+            }
             if (clan.getSquadCount() >= this.maxClanMembers) {
-                UtilMessage.simpleMessage(player, "Clans", "<alt2>Clan " + clan.getName() + "</alt2> has too many members or allies");
-                return;
+                if (!hasAdministratingClient) {
+                    UtilMessage.simpleMessage(player, "Clans", "<alt2>Clan " + clan.getName() + "</alt2> has too many members or allies");
+                    return;
+                }
+                clientManager.sendMessageToRank("Clans",
+                        UtilMessage.deserialize("<yellow>%s</yellow> was able to join <yellow>%s</yellow> because there is a player in administrative mode in their clan",
+                                player.getName(), clan.getName()),
+                        Rank.HELPER);
+
             }
         } else {
             this.clientManager.sendMessageToRank("Clans", UtilMessage.deserialize("<yellow>%s<gray> force joined <yellow>%s", player.getName(), clan.getName()), Rank.HELPER);
```
The invite data stores a gamer, we could probably check if the client who invited them is administrating instead of needing to do this

StudentAlleg replied:
It stores the inviter as the clan, not the gamer that invited them.

Mykindos committed:
Force party members to be in a safe zone

8SIXSect committed:
Woodcutting: Tree Compactor (#1058)

* Created Tree Compactor cmd

* Created Tree Compactor item

* Updated tree compactor description

* Added basic compaction functionality

* Fixed tab completions

* Allowed for tree compactor to only compact specific log types

* Implemented 5 minute cooldown for Tree Compactor

* Now compacted logs can be sold to shop keeper

* Added dependency for Tree Compactor

* Temporarily removed perk from menu

* Allowed the player to be able to see usage w/ the cooldown triggering

* Removed second line

* Changed code so that if no logs were compacted, cooldown will instantly reset (so just not happen)

* Refactored code for tab completions

* Admins can now place compacted logs

* Created new SQL migration for the compacted log item

Mykindos committed:
Fix sql name

Mykindos committed:
Update woodcutting profession tree

Mykindos committed:
Fix durability not showing on armour

Mykindos committed:
Fix rocky's migration script

Mykindos committed:
Enable flyway repair for non-mineplex databases

StudentAlleg committed:
Refactor Item Durability to use new Minecraft Functionality (#1101)

* #1092 - Update custom durability to use setMaxDamage in display. For damageable items, use vanilla damage as the display.

* #1092 - Refactor custom durability to use the vanilla system (which allows wanted behavior since 1.21). Remove CustomDamageDurabilityEvent and instead use PlayerItemDamageEvent to track damage.

* #1101 - remove unused code

---------

Co-authored-by: Braulio <40250304+braulio-dev@users.noreply.github.com>

ezrn committed:
Fix armor swap message and functionality (#1106)

* fix armor swap message and functionality

* fix

* fix2

Mykindos committed:
Fix iteminfo a little

StudentAlleg committed:
#1059 - Add a way to stop energy degeneration. Add singletons to most of ClansListeners. Explicitly reload-listeners in plugin (#1103)

StudentAlleg committed:
Check max damage before setting max damage (#1108)

Mykindos committed:
Dungeon fixes

ezrn committed:
Players drop coins on death (#1109)

* coin drops

* Clean up

---------

Co-authored-by: Mykindos <tomhoogstra@live.com.au>

Mykindos committed:
Fix misc issues

Mykindos committed:
Clean up unused

Mykindos committed:
Add back button to settings, fix misc issues

Mykindos committed:
Add soup to /kit

Mykindos committed:
silence duration changes

ezrn committed:
Ranger Bow Ability Reworks (#1070)

* bow ability changes

* bow ability fixes

* fix bow skills

* simplify logic

* fix fire

ezrn committed:
Let players /kill instantly while in safezones and while in own claim (#1112)

* slash kill

* chickenchipz fix

expires committed:
Storm sphere (#1102)

* storm sphere :fire:

* update storm sphere, ezrns idea

* update storm sphere, ezrns idea

* ezrn and samitos favourite ability

* fix changes, we ball

* ezrn and samitos favourite ability

Mykindos committed:
Misc fixes

StudentAlleg committed:
Add PvP Timer/Protection (#994)

* #942 - Work on adding pvp timer

* #942 - Add more restrictions for protection, update way to remove it, refactor common things to core

* #942 - Implement reset protection for testing purposes, remove used time on quit, add method for updating time to Gamer

* #942 - Prevent players from picking up items that they did not mine while under protection

* #942 | #994 - Fix Protection, update onExpire to take a notify boolean

* Protection Changes (throwables, custom velocity)

* Logging changes for protection

* #994 - Fix fire blast, update setprotection to minutes instead of seconds

* Update prefix from protected to protection for reminder

* #994 - Add config to enable/disable blocking players from entering other clan territories, add fire handling for protection, update bleed effect override

* #994 - Change sql additions to use migration

* Protection fixes

---------

Co-authored-by: Mykindos <tomhoogstra@live.com.au>

Mykindos committed:
Update some textures, small bug fixes & shopkeeper tweaks

Mykindos committed:
Fix weird time format bug

ezrn committed:
Fix Storm Sphere (#1115)

* fix storm sphere

* fix going through walls

* space

braulio-dev committed:
Fix/small fixes (#1114)

* Add transaction support

* Add missing logging and reduce lb overhead

ezrn committed:
Kit Command Cooldown (#1111)

* coin drops

* fix kit cooldown

* kit command simplification

Mykindos committed:
Fix kit command and heart

Mykindos committed:
Fix setprotection

ezrn committed:
Ranger New Bow Abilities (#1069)

* new bow abilities

* rework mark of the wolf

* fixes

ezrn committed:
Ranger Passive A Changes (#1071)

* passive A

* barbed arrow slowness

* fix barbed arrow config

* allowed barb tracking across multiple targets

* fixes

* fixes

ezrn committed:
Ranger Misc changes and Clean Up (#1073)

* ranger clean up

* exception fix

* fixes

---------

Co-authored-by: Mykindos <tomhoogstra@live.com.au>

ezrn committed:
Ranger Axe Changes (#1075)

* ranger axe

* fix PR

* remove feather feet

Mykindos committed:
Fix protection stopping velocity where the target is the same as the source

expires committed:
small iterator fix (#1117)

Mykindos committed:
Barbed Arrows & Kinetics passive B + minor fixes

braulio-dev committed:
Add logging to Mining Handler (#1119)

* Add transaction support

* Add missing logging and reduce lb overhead

* Add logging to MiningHandler

Mykindos committed:
Add block data to fields ores

Mykindos committed:
Fix fields not saving error

Mykindos committed:
Include some default paper configs

Mykindos committed:
Minor fixes

Mykindos committed:
Fixes & cosmetic

8SIXSect committed:
Swapped OAK_WOOD for CRIMSON_HYPHAE (#1127)

* Swapped OAK_WOOD for Crimson hyphae

* Fix

---------

Co-authored-by: Mykindos <tomhoogstra@live.com.au>

Mykindos committed:
Fix fast recovery description

StudentAlleg committed:
Update Default Configs (#1116)

* Update Clan default config

* Update Champions default config

* Update Core default config

* Clans config requested changes

Mykindos committed:
Slightly increase mage armour

Mykindos committed:
Fix migration

Mykindos committed:
Misc fixes

Mykindos committed:
Fix map being inserted when keep inventory is enabled

Fix bank subcommand

Disable some unused menu stuff

Add more items to shop

Mykindos committed:
Stop FetchNearbyEntityEvent from targeting players with protection

Mykindos committed:
FIX #1138

Mykindos committed:
Fix #1136

Mykindos committed:
Fix #1134

Mykindos committed:
Decrease blood sphere damage by from 5 to 3 per second

Mykindos committed:
Fix DB issues

Mykindos committed:
Fix auction house

Mykindos committed:
Stop players from listing blocks on the AH

Mykindos committed:
Fix coal price & fix bark being wrong model data

TR4E committed:
Update Clan.java (#1142)

Mykindos committed:
Fix bark texture & bark dupe
Fix coal price & remove compass from shop
Fix ghost claims on server restart
Fix shaman getting lured into lava

Mykindos committed:
Merge branch 'master' of https://github.com/Mykindos/BetterPvP

Mykindos committed:
Optimize Clan Core check

Mykindos committed:
Add energy at fields tip

expires committed:
changes to smelter (#1152)

Mykindos committed:
Profession level leaderboards + fix smelter

Mykindos committed:
Fix imports

Mykindos committed:
Energy from wilderness amethyst & deposit old mined amethyst

Mykindos committed:
Scale to amount

Mykindos committed:
Support mythic mobs threat tables

Mykindos committed:
Support mythic mobs threat tables + Reduce BloodSphere duration

StudentAlleg committed:
#1154 - Re-arrange transport menu to match world layout (#1159)

Mykindos committed:
Auction house fixes

Mykindos committed:
Fix woodcutting leaderboard, fix logging out to reset lives
stop crafting shulker boxes

Mykindos committed:
Fix threat table

StudentAlleg committed:
Minor description changes (#1160)

* #1126 - clarify that it has to be own melee hits for barbed arrow to work

* #1153 - Fix wind burst skill description

* #1050 - Add a the in fissure description

StudentAlleg committed:
#1149 - Fix moving item from main hand causing errors (#1151)

Mykindos committed:
Fix bark bounty in other clans chunks

Mykindos committed:
Remove crossbows from shop

Mykindos committed:
Update log sell price

Mykindos committed:
Fix shop bought items not stacking with naturally acquired

Mykindos committed:
- Increase arrow damage by 1
- Blood sphere radius no longer scales with level, but base radius is increased by 1 block
- Fire blast damage & fire duration reduced
- Tether shot radius and escape radius reduced

ItzKiba committed:
Disallow item pickups while Smoke Bomb active (#1161)

Mykindos committed:
Fix tree feller, mythic mobs stuff, stripped log stuff

Mykindos committed:
Update compacted log price

Mykindos committed:
Add dungeons settings to /settings

Mykindos committed:
Reduce silencing scaling even more

ItzKiba committed:
Vanish command (#1172)

* Disallow item pickups while Smoke Bomb active

* Add Vanish command for admins

* switch to set

* Add name to vanish effect

8SIXSect committed:
Fix Woodcutting Descriptions (#1177)

* Fix Bark Bounty desc

* Fixed woodcutting descriptions

StudentAlleg committed:
Fix corner claiming (#1171)

Mykindos committed:
Fix clone

Mykindos committed:
Fix #1163 - Tether shot

Mykindos committed:
Fix losing stuff when out of lives in dungeon

StudentAlleg committed:
Punishment Improvements (#1179)

* #839 - Prevent reason from being sent to everyone
 #840 - Prevent being able to punish without a reason

* #841 Change from .equals to .isEmpty()

* Change how punisher is retrieved, add duration to history message

StudentAlleg committed:
Fix Fire Damage Source Attribution causing incorrect kilers (#1175)

* #1170 - Fix Napalm Arrow not setting fire damage correctly

* #1169 - Remove fire damage source after duration of fire is past

ItzKiba committed:
Defensive aura changes (#1176)

* Disallow item pickups while Smoke Bomb active

* Add Vanish command for admins

* switch to set

* Fixed Defensive Aura description and added visual effect

* Remove wrong file

I forgot to make a branch so the vanish change was in this one too oops

* Pass receivers

Mykindos committed:
Fixes & hats

Mykindos committed:
Make it V6

StudentAlleg committed:
Improve Alliance feedback, fix ally bug (#1195)

* #1156 - Make ally fail messages clearer
Fix alliance logic to work on direct alliances only

* #1156 - add a check for a clan already at the max clan size trying to ally

StudentAlleg committed:
#1182 - Prevent water buckets from being placed (#1185)

Mykindos committed:
Fix store ratelimit + slight swarm buff

StudentAlleg committed:
#1120 - Fix pickaxes not losing durability in fields (#1196)

Update damaging items to use ItemStack#damage instead of trying to use custom logic to do so.

StudentAlleg committed:
Prevent attempting to damage/break air. (#1198)

StudentAlleg committed:
Show who clicked the combat log sheep (#1180)

* #1165 - Show who clicked the combat log sheep

* Fix grammar/wording for combat log

StudentAlleg committed:
#1191 - Fix poison being able to kill a player in some circumstances. Improve poison damage calculation near 2 health. (#1197)

StudentAlleg committed:
#1201 - Prevent takedown from colliding with armor stands (#1202)

Mykindos committed:
Fix auction house dupe, add energy drops to more things
Change arctic armour extending ice prison
Stop tridents from being picked up with trishot
other misc fixes

Mykindos committed:
Auction house fixes + increase immolate energy usage

Mykindos committed:
Disable longshot above 100Y

StudentAlleg committed:
#1210 - Add lore to lapis block that explains how it is used. Make it a BPvPItem (#1212)

expires committed:
Silencing Strikes and Shocking Strikes changes (#1214)

* fix dogshit assassin passives

* debuff silencing again

* debuff silencing again

StudentAlleg committed:
Fix woodcutting durability handling, add profession experience admin commands (#1208)

* #1203 - fix woodcutting/treefeller durability handling

* #1124 - Add admin commands for setting profession experience to players

Mykindos committed:
Nerf honeycomb min price

Mykindos committed:
Fix withdrawing items from mailbox with a full inventory

Mykindos committed:
Lightning orb damage no longer scales with level

StudentAlleg committed:
Fix #1221 and #1220 (#1223)

StudentAlleg committed:
Update/Fix Swarm, Defensive Stance, Sacrifice (#1217)

* #1216 - Update swarm backend to be in line with other skills

* #1068 - Fix DefensiveStance energy usage

* #1076 - Change Sacrifice to not use ceiling, damage will always be changed by the exact percent

* Remove particle method from defensive stance

StudentAlleg committed:
#1209 - Prevent leaving a clan while in combat, prevent kicking a player that is in combat from the clan (#1218)

Mykindos committed:
Feat/cosmetics (#1224)

* Add Light Manager / Disable Cosmetic Particles in combat (#1085)

* Add LightManager, which tracks a player and adds a light to their position with the specified level. Added light levels to hats

* Remove light on player logout

* Disable Viewing/Having cosmetic particles in combat

* Track when players last moved, make isMoving check for trails more accurate (#1091)

---------

Co-authored-by: StudentAlleg <36683605+StudentAlleg@users.noreply.github.com>

Mykindos committed:
Update offers

Mykindos committed:
Disable sponges from regenerating

StudentAlleg committed:
Add ignore for messages, add a way to silence global chat (#1207)

* Add /ignore, to ignore private messages

* Add a way to silence global chat

* Prevent muted players from privately messaging players

* Add config changes

* Prevent sending messages to players that have the sender ignored

* Prevent ignoring staff

* Fix incorrect package

StudentAlleg committed:
#1232 - Teleporting now cancels delayed actions (#1235)

StudentAlleg committed:
#1231 - add /logout as a temporary measure to help players know when they are safe or unsafe to logout (#1236)

Mykindos committed:
Block PDC changes

Mykindos committed:
Merge branch 'master' of https://github.com/Mykindos/BetterPvP

Mykindos committed:
final pdc try clean up and fail

Mykindos committed:
Give immolate vulnerability, fix fields bug

Mykindos committed:
Change hold pos duration scaling and make slowness 3 instead of 4

8SIXSect committed:
Fixed woodcutting dependencies (#1239)

StudentAlleg committed:
Protection Fixes (#1167)

* Protection Fixes

* Prevent using Clone with protection

* Fix pestilence activating on protected players

* #1187 - Handle protection in PreCustomDamageEvent
#1184 - Disable using tether with protection
Disable storm sphere
#1183 - Prevent rooting axe from being used on players with protection, prevent players with protection from using rooting axe
Protection Misc changes -
Reserve most items dropped or received by protected players for 10 seconds (i.e. fields, fishing, dropping, breaking). Always allow the protected player to pickup these items, even after their reservation expires.
Fishing - reserve fished up drops for 10 seconds

* Remove logging, stop tether shot from working on protected players, fix logic error with reservations

---------

Co-authored-by: Mykindos <tomhoogstra@live.com.au>

StudentAlleg committed:
#1242 - Fix Building Shopkeeper having custom model data for some items (#1243)

Mykindos committed:
Fix immolate

Mykindos committed:
Update hold pos config

Mykindos committed:
Update immolate description

Mykindos committed:
Adjust spawn rates & default view distance

Mykindos committed:
Rake puts items in inventory + make immolate vuln 2

Mykindos committed:
Cancel auctions when core is destroyed

Mykindos committed:
Minor optimization

StudentAlleg committed:
#1240 - FIx CrossbowWeaponTip being empty (#1245)

Mykindos committed:
Shops are no longer safe for pillaging/pillaged clans

Mykindos committed:
Add playtime command

Mykindos committed:
Allow players to ride minecarts again

Mykindos committed:
Dont reduce size while base fishing

Mykindos committed:
Fix combat in safezone

Mykindos committed:
Fix energy dupe

StudentAlleg committed:
#1211 - Give temporary immunity from being combat tagged after death (#1251)

StudentAlleg committed:
#1181 - Prevent imbued items from being sold (#1246)

Pull up no shop sell namespacedkey to core

Mykindos committed:
Despawn fish after 1 minute

StudentAlleg committed:
#1186 - Use completeable futures to fetch punisher name (#1199)

Mykindos committed:
Fix #1254

Mykindos committed:
Fix error disbanding clan

Mykindos committed:
Fix kill credit

Mykindos committed:
Fix bark dupe

Mykindos committed:
Update profession config + add cherry log to xp table

Mykindos committed:
Fix clan energy bug

StudentAlleg committed:
#1241 - Prevent runes from being stacked (#1262)

Mykindos committed:
Trail fixes, shop fixes + more shop items

Mykindos committed:
Also remove dropped fish after a minute

8SIXSect committed:
Updated woodcutting to support cherry trees (#1264)

* Updated woodcutting to support cherry trees

* Updated tree feller config

* No change in height

StudentAlleg committed:
#1267 - Change how setprotection gets player (#1269)

StudentAlleg committed:
Fix deflection base charges (#1261)

* Fix deflection base charges

* Remove sl4j logging

* Remove logging import

* Fix deflection maxLevel

StudentAlleg committed:
#1237 - Allow triggering tripwires in all circumstances (#1265)

#1178 - Allow crafting table use in admin clans

Mykindos committed:
Optimisations and misc fixes

Mykindos committed:
Store player name changes, Fix #1213

Mykindos committed:
Make rupture armour stands non-persistent

Mykindos committed:
Stop big trees from growing

ItzKiba committed:
Three warlock effects (#1206)

* Disallow item pickups while Smoke Bomb active

* Add Vanish command for admins

* switch to set

* Fixed Defensive Aura description and added visual effect

* Remove wrong file

I forgot to make a branch so the vanish change was in this one too oops

* Pass receivers

* Three warlock AOE effects

Mykindos committed:
Fix growth of other big trees

Mykindos committed:
Fix error with protection picking up items

Mykindos committed:
Fix #1194 - Fix napalm dealing damage in water, same with immolate

Mykindos committed:
Stop mushroom spreading outside of unlocked farming levels

Mykindos committed:
Adjust mushroom / glowberry growth, make soup dynamic & add glowberry to shop

Mykindos committed:
Fix spread logic

Mykindos committed:
Cache player heads + show world size

Mykindos committed:
Fix aegis durability

Mykindos committed:
Fixed an issue where redstone related blocks could be placed on the edge of a claim to open doors

Mykindos committed:
Remove old code

StudentAlleg committed:
Prevent inviting a player to a clan or joining a clan that would cause allies to have an illegal squad count. (#1266)

Mykindos committed:
Fix redstone thing again & dungeon lore

Mykindos committed:
Misc fixes

Mykindos committed:
Fix #1295

Mykindos committed:
Fix bonemealing mushrooms

8SIXSect committed:
Cherry logs now work w/ treecompactor (#1287)

Mykindos committed:
Fix tree feller

StudentAlleg committed:
Prevent block toss from colliding with spectators (#1303)

* #1222 - Prevent block toss from colliding with spectators

* Remove left over test logging

StudentAlleg committed:
#1286 - Add creative only functionality to /treecompactor (and disable normal functionality), prevent treecompactor from filling a creative players inventory because items are not removed from a creative player (#1300)

StudentAlleg committed:
#1168 - Prevent loading a lootable with a disabled item (#1301)

StudentAlleg committed:
#1292 - Fix ice prison replacing some redstone items (#1297)

Pulled restone block logic to utilblock

StudentAlleg committed:
#1296 - Remove block break warning, make reserve time configurable (#1299)

StudentAlleg committed:
#1188 - Make bleed damage clarify that it is true damage (#1302)

ezrn committed:
Change c stuck timer to the same duration as c home (#1107)

* changed c stuck timer to 20 seconds

* 30 seconds

Mykindos committed:
Stop trishot working with longshot & other optimizations

Mykindos committed:
Fix #1285

Mykindos committed:
Fix cache, add YT rank, optimize

StudentAlleg committed:
Add water block tip (#1304)

* #1225 - Add water block tip

* Remove old logging from testing

Mykindos committed:
Lil optimizations

Mykindos committed:
Bigger optimization

Mykindos committed:
Make YT prefix bold

Mykindos committed:
Fix #1310

Mykindos committed:
Fix issue where original owner was getting set even if the pick up event was cancelled

Mykindos committed:
Fix clan kick command

Mykindos committed:
Can no longer enemy admin clans, added ability to remove a clans admin status

Mykindos committed:
Adjust default damage indicator size

Mykindos committed:
Add sharkbait, a new fishing legendary

Mykindos committed:
Fix issue with /c demote and /c promote, add index to client name

Mykindos committed:
Fix major bugs

Mykindos committed:
Fix protection issue & hopeful leaderboard fix

Mykindos committed:
Fix issues, make hyper fast at log cutting

StudentAlleg committed:
#1307 - prevent damaging self with restore blocks (#1315)

StudentAlleg committed:
Fix Artic Armor giving slowness on non-melee hits (#1314)

StudentAlleg committed:
Add fall damage immunity when using sponges (#1305)

* #1293 - Add fall damage immunity when using sponges

* #1305 - Expand isStandingOnCheck

Mykindos committed:
Put a cap on tips command

Mykindos committed:
New events, bug fixes

Mykindos committed:
Fixes

Mykindos committed:
Fix fall damage combat tagging you

Mykindos committed:
Fix damage tint & misc errors

Mykindos committed:
Max core y limit

Mykindos committed:
Remove old energy on swing code

Mykindos committed:
Fixes + make arrow damage apply to all arrows

Mykindos committed:
Nerf combo attack duration & fix spectral arrow damage

Mykindos committed:
Import cleanup & optimization

Mykindos committed:
Force async combat stats

Mykindos committed:
Fix weird wall kick exception

Mykindos committed:
Add indexes to kills & kill contributions

Mykindos committed:
Add punishment index

StudentAlleg committed:
#1324 - Update vault-restrictions.yml to not include an extra hyper-axe (#1325)

Mykindos committed:
Provide some evade relief

Mykindos committed:
Adjust

Mykindos committed:
Add woodcutting and fishing indexes

Mykindos committed:
Delay leaderboard updates & add new index

Mykindos committed:
Only insert compacted log if the correct amount was removed

Mykindos committed:
Remove targets player cannot see from clone

expires committed:
Fix Clone (#1345)

* fix clone targetting itself LOL

* fix clone targetting itself LOL

* fix it from targetting teammates clones

StudentAlleg committed:
Fix mediaChannelOptional not actually being used to set default value if not present (#1339)

StudentAlleg committed:
Improve Vanish (#1334)

* #1333 - Improve Vanish. Explicity remove players from tab list for commandVanish, add fake join/leave messages for commandVanish, show commandVanished players to staff. Run vanish logic for players that join the server.
Run onReceive for players that join the server with active effects

* #1334 - Fix sonar cloud issues

expires committed:
Combo Attack Fixes (#1344)

* make combo semi decent again

* Remove useless imports

* Remove useless import

* fix getLevel call

* remove redundant import statement

Mykindos committed:
Fix #1323 - Save direction when setting home

StudentAlleg committed:
Explicitly handle null names and UUIDs in offline/advanced search (#1321)

Polisvh committed:
Threating Shout Nerf (#1332)

* Update ThreateningShout.java

* Update skills.yml

* Update ThreateningShout.java

* Update skills.yml

StudentAlleg committed:
#1326 - Fix bloodlust not using the correct levels for effects (#1331)

Mykindos committed:
Gut evade duration

vrqz committed:
Fix evade duration truncating (#1351)

Co-authored-by: ee0w <qqqajaja05@gmail.com>

Mykindos committed:
Move unstoppable force to knight

ezrn committed:
Magnetic Axe (#1000)

* test

* roped axe progress

* test

* axe done

* teleport fix

* update description

* fix error

* fix magnetic axe

Mykindos committed:
Remove Power Chop + Fix Magnetic Axe issues

Mykindos committed:
Update seismic slam

Mykindos committed:
Fix minor issues

braulio-dev reviewed (changes requested):
Just make `Clan#getSquadCount` not count clients in `/client admin`, pass ClientManager into the constructor
